crates/types/src/contract_payloads/tob/mod.rs

```diff
@@ -238,3 +245,66 @@ mod test {
         );
     }
 }
+
+/// Computes the reward checksum for a given range of ticks.
+///
+/// `from_above`: `true` = high to low, `false` = low to high.
+/// total ticks should be the length of the reward vec
+fn compute_reward_checksum(
+    start_tick: i32,
+    start_liquidity: u128,
+    snapshot: &PoolSnapshot,
+    from_above: bool,
+    mut total_ticks: usize
+) -> U160 {
+    let mut reward_checksum = [0u8; 32];
+    let mut tick = start_tick;
+
+    // Get initial liquidity directly from snapshot
+    let mut liquidity = snapshot
+        .liquidity_at_tick(start_tick)
+        .unwrap_or(start_liquidity);
+
+    loop {
+        // we only checksum the ticks that w are rewarding.
+        total_ticks -= 1;
+        if total_ticks == 0 {
+            break;
+        }
+
+        let tick_bytes: [u8; 3] = I24::try_from(tick).unwrap().to_be_bytes();
+        let hash_input =
+            [reward_checksum.as_slice(), &liquidity.to_be_bytes(), &tick_bytes].concat();
+        reward_checksum = *keccak256(&hash_input);
+
+        if from_above {
+            // Move to the next initialized tick while enforcing tick spacing
+            if let Some(next_tick) = snapshot.get_next_tick_lt(tick) {
+                tick = next_tick;
+
+                // Update liquidity for the new tick
+                liquidity = snapshot
+                    .liquidity_at_tick(tick)
+                    .expect("liquidity at tick must be defined for this tick");
+            } else {
+                break;
```

Comment:
instead of breaking here and returning a known broken checksum, lets propagate the error up as otherwise its misleading as it implies we have a bug in the code.